gradio/helpers.py

```diff
@@ -139,23 +139,23 @@ def __init__(
             )
 
         if cache_examples is None:
-            if cache_examples_env := os.getenv("GRADIO_CACHE_EXAMPLES"):
-                if cache_examples_env.lower() == "true":
-                    if fn is not None and outputs is not None:
-                        self.cache_examples = True
-                    else:
-                        self.cache_examples = False
-            elif utils.get_space() and fn is not None and outputs is not None:
-                self.cache_examples = True
```

Comment:
I also got rid of this logic and instead set the `GRADIO_CACHE_EXAMPLES` env in the Spaces Dockerfile: https://github.com/huggingface-internal/workloads/pull/2379